velox/functions/sparksql/ToJson.h

```diff
@@ -0,0 +1,333 @@
+template <>
+void toJson<TypeKind::BOOLEAN>(
+    const exec::GenericView& input,
+    std::string& result,
+    const JsonOptions& options) {
+  auto value = input.castTo<bool>();
+  static const char TRUE[] = "true";
+  static const char FALSE[] = "false";
+  result.append(value ? TRUE : FALSE);
+}
+
+template <>
+void toJson<TypeKind::TINYINT>(
+    const exec::GenericView& input,
+    std::string& result,
+    const JsonOptions& options) {
+  auto value = input.castTo<int8_t>();
+  append(value, result);
+}
+
+template <>
+void toJson<TypeKind::SMALLINT>(
+    const exec::GenericView& input,
+    std::string& result,
+    const JsonOptions& options) {
+  auto value = input.castTo<int16_t>();
+  append(value, result);
+}
+
+template <>
+void toJson<TypeKind::INTEGER>(
+    const exec::GenericView& input,
+    std::string& result,
+    const JsonOptions& options) {
+  auto value = input.castTo<int32_t>();
+  if (input.type()->isDate()) {
+    result.append("\"").append(DATE()->toString(value)).append("\"");
+  } else {
+    append(value, result);
+  }
+}
+
+template <>
+void toJson<TypeKind::BIGINT>(
+    const exec::GenericView& input,
+    std::string& result,
+    const JsonOptions& options) {
+  auto value = input.castTo<int64_t>();
+  if (input.type()->isDecimal()) {
+    auto [precision, scale] = getDecimalPrecisionScale(*input.type());
+    const size_t maxSize = DecimalUtil::maxStringViewSize(precision, scale);
+    char buffer[maxSize];
+    size_t len = DecimalUtil::castToString(value, scale, maxSize, buffer);
+    result.append(buffer, len);
+  } else {
+    append(value, result);
+  }
+}
+
+template <>
+void toJson<TypeKind::HUGEINT>(
+    const exec::GenericView& input,
+    std::string& result,
+    const JsonOptions& options) {
+  auto value = input.castTo<int128_t>();
+  const size_t maxSize = folly::detail::digitsEnough<uint128_t>() + 1;
+  char buffer[maxSize];
+  size_t p;
+  if (value < 0) {
+    *buffer = '-';
+    p = 1 +
+        folly::detail::unsafeTelescope128(buffer + 1, buffer + maxSize, -value);
+  } else {
+    p = folly::detail::unsafeTelescope128(buffer, buffer + maxSize, value);
+  }
+  result.append(buffer, p);
+}
+
+template <>
+void toJson<TypeKind::REAL>(
+    const exec::GenericView& input,
+    std::string& result,
+    const JsonOptions& options) {
+  auto value = input.castTo<float>();
+  append(value, result);
+}
+
+template <>
+void toJson<TypeKind::DOUBLE>(
+    const exec::GenericView& input,
+    std::string& result,
+    const JsonOptions& options) {
+  auto value = input.castTo<double>();
+  append(value, result);
+}
+
+template <>
+void toJson<TypeKind::VARCHAR>(
+    const exec::GenericView& input,
+    std::string& result,
+    const JsonOptions& options) {
+  auto value = input.castTo<Varchar>();
+  const size_t length = normalizedSizeForJsonCast(value.data(), value.size());
+  char buffer[length];
+  normalizeForJsonCast(value.data(), length, buffer);
```

Comment:
Consider changing `length` to `value.size()` here.

The second parameter of `normalizedSizeForJsonCast` expects the length of the input, but currently, `length` refers to the output length.

For UTF-8 characters with lengths of 1-3 bytes, input and output lengths match. However, for 4-byte UTF-8 characters (e.g., Emoji `😄`), the output length becomes 12 (e.g., `\uD83D\uDE04`).